src/renderer/base/thread.cpp

```diff
@@ -158,26 +160,46 @@ DWORD WINAPI RenderThread::_ThreadProc()
     while (_fKeepRunning)
     {
         WaitForSingleObject(_hPaintEnabledEvent, INFINITE);
-        WaitForSingleObject(_hEvent, INFINITE);
+
+        // Skip waiting if next frame is requested.
+        if (_fNextFrameRequested.test_and_set(std::memory_order_relaxed))

```

Comment:
I think this can be relaxed since it happens mostly during when the render thread is sleeping.